src/librustc_save_analysis/lib.rs

```diff
@@ -497,7 +497,7 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
     }
 
     pub fn get_path_data(&self, id: NodeId, path: &ast::Path) -> Option<Data> {
-        let def = self.tcx.expect_def(id);
+        let def = option_try!(self.tcx.expect_resolution(id).maybe_full_def());
```

Comment:
I could, but it would be (IMO) a bit of an abstraction violation - `depth` and whether a def is 'baked' or not seem like implementation details that we shouldn't be dealing with here. Plus the method seems like a thing we'd might want - there are similar methods all over the ty ctxt, etc.